include/ap/argument_parser.hpp

```diff
@@ -336,69 +430,153 @@ class argument_parser {
+
+        for (const auto& argument : parser._optional_args)
+            os << "\t" << *argument << std::endl;
+
         return os;
     }
 
+#ifdef AP_TESTING
+    friend struct ::ap_testing::argument_parser_test_fixture;
+#endif
+
 private:
-    void _check_arg_name_present(const std::string_view name) const {
-        const auto name_eq_predicate = [name](const detail::argument_interface& arg) {
-            return name == arg.name();
+    using cmd_argument_list = std::vector<std::string>;
+    using cmd_argument_list_iterator = typename cmd_argument_list::const_iterator;
+
+    using argument_ptr_type = std::unique_ptr<detail::argument_interface>;
+    using argument_opt_type =
+        std::optional<std::reference_wrapper<detail::argument_interface>>;
+    using argument_list_type = std::vector<argument_ptr_type>;
+    using argument_list_iterator = typename argument_list_type::iterator;
+
+    auto _name_eq_predicate(std::string_view name) const {
+        return [name](const argument_ptr_type& arg) {
+            return name == arg->name();
         };
+    }
+
+    void _check_arg_name_present(std::string_view name) const {
+        const auto predicate = this->_name_eq_predicate(name);
 
         if (std::find_if(
-                this->_positional_args.begin(), this->_positional_args.end(), name_eq_predicate
+                this->_positional_args.begin(), this->_positional_args.end(), predicate
             ) != this->_positional_args.end()) {
-            throw std::invalid_argument(
-                "Argument name [" + std::string(name) + "] invalid! Name already used in another argument"
-            );
+            throw std::invalid_argument("[_check_arg_name_present(n)] TODO: msg");
         }
 
         if (std::find_if(
-                this->_optional_args.begin(), this->_optional_args.end(), name_eq_predicate
+                this->_optional_args.begin(), this->_optional_args.end(), predicate
             ) != this->_optional_args.end()) {
-            throw std::invalid_argument(
-                "Argument name [" + std::string(name) + "] invalid! Name already used in another argument"
+            throw std::invalid_argument("[_check_arg_name_present(n, s)] TODO: "
+                                        "msg");
+        }
+    }
+
+    [[nodiscard]] cmd_argument_list _process_input(int argc, char* argv[]) const {
+        /*
+        TODO:
+        * assignment character detection
+        * quotes detection
+        * negative number detection
+        * split into smaller functions
+        */
+
+        if (argc < 2)
+            return cmd_argument_list{};
+
+        cmd_argument_list args;
+        args.reserve(argc - 1);
+
+        for (int i = 1; i < argc; i++) {
+            std::string arg = argv[i];
+
+            if (arg.length() > this->_flag_prefix_length and
+                arg.starts_with(this->_flag_prefix)) {
+                arg.erase(0, this->_flag_prefix_length);
```

Comment:
<img width="436" alt="image" src="https://github.com/SpectraL519/cpp-ap/assets/91973266/397b70d0-2565-4f60-b849-a15613ccbadf">
Here in line 499 could be added tabulation for clearer readability.

Reply:
That would have to be adjusted in clang-format. You can look into it :)